toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -1342,6 +1363,41 @@ func provisionUserSSHCerts(installChroot *safechroot.Chroot, user configuration.
 		if err != nil {
 			return
 		}
+
+		logger.Log.Infof("Adding ssh key (%s) to user (%s) .ssh/authorized_users", filepath.Base(pubKey), user.Name)
+		pubKeyData, err = file.ReadLines(pubKey)
+		if err != nil {
+			logger.Log.Warnf("Failed to read from SSHPubKey : %v", err)
+			return
+		}
+
+		// Append to the tmp/authorized_users file
+		for i, s := range pubKeyData {
+			_ = i
+			s += "\n"
+			err = file.Append(s, authorizedKeysTempFile)
+			if err != nil {
+				logger.Log.Warnf("Failed to append to %s : %v", authorizedKeysTempFile, err)
+				return
+			}
+		}
+	}
+
+	fileToCopy := safechroot.FileToCopy{
+		Src:  authorizedKeysTempFile,
+		Dest: authorizedKeysFile,
+	}
+
+	err = installChroot.AddFiles(fileToCopy)
+	if err != nil {
+		return
+	}
+
+	// Clean up temp file
+	err = os.Remove(authorizedKeysTempFile)
```

Comment:
Please turn it to a deferred call right after a successful creation of the file.